evalml/tests/data_checks_tests/test_data_check_action_option.py

```diff
@@ -0,0 +1,378 @@
+import re
+
+import pytest
+
+from evalml.data_checks import DataCheckActionCode, DataCheckActionOption
+
+
+@pytest.fixture
+def dummy_data_check_name():
+    return "dummy_data_check_name"
+
+
+def test_data_check_action_option_attributes(dummy_data_check_name):
+    data_check_action_option = DataCheckActionOption(
+        DataCheckActionCode.DROP_COL, dummy_data_check_name
+    )
+    assert data_check_action_option.data_check_name == dummy_data_check_name
+    assert data_check_action_option.action_code == DataCheckActionCode.DROP_COL
+    assert data_check_action_option.metadata == {"rows": None, "columns": None}
+    assert data_check_action_option.parameters is None
+
+    data_check_action_option = DataCheckActionOption(
+        DataCheckActionCode.DROP_COL, None, metadata={}, parameters={}
+    )
+    assert data_check_action_option.action_code == DataCheckActionCode.DROP_COL
+    assert data_check_action_option.data_check_name is None
+    assert data_check_action_option.metadata == {"rows": None, "columns": None}
+    assert data_check_action_option.parameters == {}
+
+    parameters = {
+        "global_parameter_name": {
+            "parameter_type": "global",
+            "type": "float",
+            "default_value": 0.0,
+        },
+        "column_parameter_name": {
+            "parameter_type": "column",
+            "columns": {
+                "a": {
+                    "impute_strategy": {
+                        "categories": ["mean", "mode"],
+                        "type": "category",
+                        "default_value": "mean",
+                    },
+                    "constant_fill_value": {"type": "float", "default_value": 0},
+                },
+            },
+        },
+    }
+    data_check_action_option = DataCheckActionOption(
+        DataCheckActionCode.DROP_COL,
+        dummy_data_check_name,
+        metadata={"columns": [1, 2]},
+        parameters=parameters,
+    )
+    assert data_check_action_option.action_code == DataCheckActionCode.DROP_COL
+    assert data_check_action_option.data_check_name == dummy_data_check_name
+    assert data_check_action_option.metadata == {"columns": [1, 2], "rows": None}
+    assert data_check_action_option.parameters == parameters
+
+
+def test_data_check_action_option_equality(dummy_data_check_name):
+    data_check_action_option = DataCheckActionOption(
```

Comment:
I think the equality or inequality tests cover every case but it would be nice to double check since code coverage won't catch the cases because of how the loop is written!